src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -348,6 +392,37 @@ private void EnsureSignature()
             }
         }
 
+        private void EnsureAccessorsBinding()
+        {
+            if (!_state.HasComplete(CompletionPart.FinishPropertyAccessorsBinding))
+            {
+                if (_state.NotePartComplete(CompletionPart.StartPropertyAccessorsBinding))
+                {
+                    // Ensure the binding is done so we guarantee that we have the BackingField set if necessary.
+                    // If we already have a backing field. We should still do binding since we create a backing field just if
+                    // we see an initializer. In this case binding is needed to know whether it's an auto property. See CreateBackingField method.
```

Comment:
>In this case binding is needed to know whether it's an auto property. [](http://example.com/codeflow?start=46&length=69)

Regarding:
```
// If we already have a backing field. We should still do binding since we create a backing field just if
// we see an initializer. In this case binding is needed to know whether it's an auto property. See CreateBackingField method.
```

This behavior around initializers is very unfortunate. I see the problem it creates. Specifically the fact that the created backing field doesn't have ```IsCreatedForFieldKeyword``` flag set when there is an initializer, there is no semicolon-only accessor, and there is a usage of a ```field``` keyword. Essentially, in that situation we have a baking field symbol that "lies" to us. I think we should definitely address that.

This isn't a complete solution, but here is the direction I think we should explore:
I am assuming that having an initializer, not having a semicolon-only accessor and not having a ```field``` keyword is an error. I also think that creation of the backing field based solely on presence of an initializer is just an error recovery mechanism, we need a field symbol so that SemanticModel can bind the initializer. Given that, I think that in a situation when we don't have a semicolon-only accessor, but have an initializer, we should create a backing field with ```IsCreatedForFieldKeyword``` flag optimistically set to true. BTW, I am fine with ```IsCreatedForFieldKeyword``` == true in this error scenario as apposed to ```IsCreatedForFieldKeyword``` == false in presence of the ```field``` keyword and absence of a semicolon-only accessor, which is a success scenario. Later on, we obviously need to confirm that there is indeed a ```field``` keyword usage and report an error if there isn't one. This is the part that requires additional thinking to figure out where is the right place to perform the check and what would be the mechanism to confirm the actual presence of the ```field``` keyword. Right now I am leaning towards recording this fact as a separate flag in ```SynthesizedBackingFieldSymbol``` (there is still question when to do that). It is very likely that this flag is going to be required anyway, because we want to be able to distinguish ```SynthesizedBackingFieldSymbol```s that were created from speculative binding through SemanticModel and those that correspond to "real" ```field``` keywords in the original source.